mwdb/web/src/components/UserPasswordRecover.js

```diff
@@ -1,124 +1,111 @@
-import React, { Component } from "react";
+import React, { useState, useContext } from "react";
 import ReCAPTCHA from "react-google-recaptcha";
 
 import api from "@mwdb-web/commons/api";
 import { ConfigContext } from "@mwdb-web/commons/config";
 import { View } from "@mwdb-web/commons/ui";
 
-export default class UserPasswordRecover extends Component {
-    constructor(props) {
-        super(props);
-        this.state = this.initialState;
-    }
-
-    static contextType = ConfigContext;
-
-    get initialState() {
-        return {
-            login: "",
-            email: "",
-            success: false,
-            dirty: false,
-            error: null,
-            recaptcha: null,
-        };
-    }
+export default function UserPasswordRecover() {
+    const initialState = {
+        login: "",
+        email: "",
+        success: false,
+        error: null,
+        recaptcha: null,
+    };
 
-    componentDidMount() {
-        if (this.dirty) this.setState(this.initialState);
-    }
+    const config = useContext(ConfigContext);
+    const [state, setState] = useState(initialState);
 
-    handleInputChange = (event) => {
+    const handleInputChange = (event) => {
         const target = event.target;
         const value =
             target.type === "checkbox" ? target.checked : target.value;
         const name = target.name;
 
-        this.setState({
+        setState({
+            ...state,
```

Comment:
If we derive from a state, we need to use a callback to make that change synchronous with other changes

Don't apply this suggestion directly, because it lacks of closing parenthesis
```suggestion
        setState(state => ({
            ...state,
```